GameDataParser/Program.cs

```diff
@@ -1,33 +1,61 @@
-﻿using System.Globalization;
+﻿using System;
+using System.Threading;
+using System.Globalization;
 using System.IO;
 using System.IO.MemoryMappedFiles;
 using System.Xml;
+using System.Collections.Generic;
 using GameDataParser.Crypto;
 using GameDataParser.Crypto.Stream;
 using GameDataParser.Files.Export;
+using GameDataParser.Crypto.Common;
+using GameDataParser.Files;
 
 namespace GameDataParser
 {
     internal static class Program
     {
-        public static ItemMetadataExport Item = new ItemMetadataExport();
-        public static MapMetadataExport MapEntity = new MapMetadataExport();
-        public static SkillMetadataExport Skill = new SkillMetadataExport();
-        public static InsigniaMetadataExport Insignia = new InsigniaMetadataExport();
-        public static PrestigeMetadataExport Prestige = new PrestigeMetadataExport();
-        public static ExpMetadataExport ExpTable = new ExpMetadataExport();
-
         private static void Main()
         {
             // Force Globalization to en-US because we use periods instead of commas for decimals
             CultureInfo.CurrentCulture = new CultureInfo("en-US");
 
-            Item.Export();
-            MapEntity.Export();
-            Skill.Export();
-            Insignia.Export();
-            Prestige.Export();
-            ExpTable.Export();
+            // Resources
+            string xmlHeaderFile = VariableDefines.XML_PATH.Replace(".m2d", ".m2h");
+            string exportedHeaderFile = VariableDefines.EXPORTED_PATH.Replace(".m2d", ".m2h");
+
+            List<PackFileEntry> xmlFiles = FileList.ReadFile(File.OpenRead(xmlHeaderFile));
+            List<PackFileEntry> exportedFiles = FileList.ReadFile(File.OpenRead(exportedHeaderFile));
+
+            MemoryMappedFile xmlMemFile = MemoryMappedFile.CreateFromFile(VariableDefines.XML_PATH);
+            MemoryMappedFile exportedMemFile = MemoryMappedFile.CreateFromFile(VariableDefines.EXPORTED_PATH);
+
+            // Threads
+            Thread itemThread = new Thread(() => ItemMetadataExport.Export(xmlFiles, xmlMemFile));
+            Thread mapEntityThread = new Thread(() => MapMetadataExport.Export(exportedFiles, exportedMemFile));
+            Thread skillThread = new Thread(() => SkillMetadataExport.Export(xmlFiles, xmlMemFile));
+            Thread insigniaThread = new Thread(() => InsigniaMetadataExport.Export(xmlFiles, xmlMemFile));
+            Thread prestigeThread = new Thread(() => PrestigeMetadataExport.Export(xmlFiles, xmlMemFile));
+            Thread expThread = new Thread(() => ExpMetadataExport.Export(xmlFiles, xmlMemFile));
```

Comment:
You should just create an abstract class that all these exporters extend from so you can define these once and then do 3 for loops, one to make these threads, another to start them, then finally to join them

Reply:
Can you give an example of how it would loop I'm not quite understanding, are you saying I should make a list of all the export class objects and then iterate over a loop calling export?

Reply:
Nevermind, MapMetadataExport takes in different parameters, which makes changing this not as useful